src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -73,5 +73,17 @@ public class ToolSettings
         /// </code>
         /// </example>
         public IDictionary<string, string> EnvironmentVariables { get; set; } = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the exit code from the tool process causes an exception to be thrown.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// DotNetCoreTest("MyProject.csproj", new DotNetCoreTestSettings {
+        ///     IgnoreExitCode = true
+        /// });
+        /// </code>
+        /// </example>
+        public bool IgnoreExitCode { get; set; }
```

Comment:
I'm wondering if a more flexible way would be
```suggestion
        public Func<int, bool> HandleExitCode { get; set; }
```
Then ignoring it would be as simple as
```csharp
HandleExitCode = _=> true
```
but it would also allow more granual control too i.e.
```csharp
HandleExitCode = exitCode => exitCode switch {
                                1 => true, // threat 1 and 2 as handled "ok".
                                2 => true,
                                _ => false
                             }
```

Tool implementation could then be something like
```csharp
var exitCode = process.GetExitCode();
if (!settings.HandleExitCode?.Invoke(exitCode) ?? true)
{
    ProcessExitCode(exitCode);
}
```

Reply:
I like it. I'll update it to do that.

Reply:
@flcdrg excellent, thanks! Just ping when you've addressed feedback and rebased against latest develop.

Reply:
@devlead rebased and ready for review :-)